File: Exercise/LuoGu/Code_2023_7_18/P1075.java
package Code_2023_7_18;

import java.util.Scanner;

public class P1075 {

    public static boolean isPrime(int n) {
        int factor_min = (int) Math.sqrt(n);
        for (int i = 2; i <= factor_min; i++) {
            if (n / i * i == n) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int factor_min = (int) Math.sqrt(n);
        int[] prime_arr = new int[factor_min];
        int index = 0;

        for (int i = 2; i <= factor_min; i++) {
            if (isPrime(i)) {
                prime_arr[index++] = i;
            }
        }
        for (int i = 0; i < index; i++) {
            if (n / prime_arr[i] * prime_arr[i] == n) {
                System.out.println(n / prime_arr[i]);
                break;
            }
        }
        sc.close();
    }
}

File: Exercise/LuoGu/Code_2023_7_18/P1075.cpp
#include <iostream>
#include <algorithm>
#include <math.h>
#include <vector>
#include <string>

using namespace std;
bool is_prime(int n)
{
    int factor_min = (int)sqrt(n);
    for (int i = 2; i <= factor_min; i++)
    {
        if (n / i * i == n)
        {
            return false;
        }
    }
    return true;
}
int main(void)
{
    int n;
    cin >> n;
    int factor_min = (int)sqrt(n);

    int prime_arr[factor_min];
    int index = 0;

    for (int i = 2; i <= factor_min; i++)
    {
        if (is_prime(i))
        {
            prime_arr[index++] = i;
        }
    }

    for (int i = 0; i < index; i++)
    {
        if (n / prime_arr[i] * prime_arr[i] == n)
        {
            cout << n / prime_arr[i] << endl;
            break;
        }
    }

    return 0;
}